### DISCUSSION TITLE
use-form hooks with local storage behavior

### DISCUSSION THREAD

**Discussion by matheusvellone** [mantinedev/mantine#1319] [Feature requests]
Hi, before anything thanks for this amazing lib!

I'm currently developing a personal dashboard and thought that might be useful to have a form where some fields are automatically synced with local storage but in the current version of the lib (v4.1.5) I could not find a way to do it.

Thought about it a little, and came up with this API
```js
const form = useForm({
    initialValues: {
        user: '', // this value would be used if the key doesn't exist in local storage
        password: '',
    },
    useLocalStorage: { // or maybe an array passing keys to enable localstorage on them
        user: true,
    },
  })
```

I'd be happy to code this if you think its a good idea.

**Comment by rtivital** [comment]
I have plans to add local storage but in other form. Currently you can use useEffect hook to save/retrive your values from local storage.
